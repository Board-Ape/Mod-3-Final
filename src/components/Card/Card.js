import React from 'react';
import PropTypes from 'prop-types';
import { Link } from 'react-router-dom';
import '../App/App.css';

const Card = ({house}) => {
  return (
    <Link to={'/'}>
      <div className='card'>
        <h1>{house.name}</h1>
        <h2>FOUNDED: {house.founded}</h2>
        <h3>SEATS: {house.seats.map(seats => {
          return seats;
        })}</h3>
        <h4>COAT OF ARMS: {house.coatOfArms}</h4>
        <h4>ANCESTRAL WEAPONS: {house.ancestralWeapons.map(weapons => {
          return weapons;
        })}</h4>
        <h1>{house.words}</h1>
      </div>
    </Link>
  );
};

Card.propTypes = {
  house: PropTypes.object
};

export default Card;
