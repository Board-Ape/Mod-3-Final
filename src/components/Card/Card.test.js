import React from 'react';
import Card from '../Card/Card';
import { shallow } from 'enzyme';
import mockData from './mockData';

describe('Card Tests', () => {

  it('should render component and match snapshot', () => {
    const renderedCard = shallow(
      <Card house={mockData[0]} />);

    expect(renderedCard).toMatchSnapshot();
  });
});
