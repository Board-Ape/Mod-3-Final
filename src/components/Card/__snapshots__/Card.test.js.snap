// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card Tests should render component and match snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Card
    house={
        Object {
            "ancestralWeapons": Array [
              "Lady Forlorn",
            ],
            "coatOfArms": "Three black...",
            "diedOut": "",
            "founded": "",
            "name": "House Corbray of Heart's Home",
            "seats": Array [
              "Heart's Home",
            ],
            "titles": Array [
              "Lord of Heart's Home",
              "King of the Fingers (historical)",
            ],
          }
    }
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {},
    "ref": null,
    "rendered": null,
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {},
      "ref": null,
      "rendered": null,
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactFifteenAdapter {
      "options": Object {
        "supportPrevContextArgumentOfComponentDidUpdate": true,
      },
    },
  },
}
`;
